qualtran/bloqs/comparison_gates.ipynb

```diff
@@ -0,0 +1,957 @@
+    "\n",
+    "Note that as in the case of 2 qubits we still need a final `SingleQubitCompare` operation to extract the result."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def quantum_quantum_comparator(\n",
+    "    A: Sequence[cirq.Qid],\n",
+    "    B: Sequence[cirq.Qid],\n",
+    "    z: cirq.Qid,\n",
+    "    less_than: cirq.Qid,\n",
+    "    greater_than: cirq.Qid,\n",
+    "):\n",
+    "    adjoint = []\n",
+    "    anc_cnt = 0\n",
+    "\n",
+    "    def build_tree(A: Sequence[cirq.Qid], B: Sequence[cirq.Qid]) -> cirq.OP_TREE:\n",
+    "        nonlocal adjoint, anc_cnt\n",
+    "        if len(A) <= 1:\n",
+    "            return\n",
+    "        if len(A) == 2:\n",
+    "            anc = [cirq.NamedQubit(f'anc_{i+anc_cnt}') for i in range(3)]\n",
+    "            anc_cnt += 3\n",
+    "            op = cirq_ft.algos.BiQubitsMixer().on_registers(x=A, y=B, ancilla=anc)\n",
+    "            yield op\n",
+    "            adjoint.append(op**-1)\n",
+    "            return\n",
+    "        m = len(A) >> 1\n",
+    "        yield from build_tree(A[:m], B[:m])\n",
+    "        yield from build_tree(A[m:], B[m:])\n",
+    "\n",
+    "        anc = [cirq.NamedQubit(f'anc_{i+anc_cnt}') for i in range(3)]\n",
+    "        anc_cnt += 3\n",
+    "        op = cirq_ft.algos.BiQubitsMixer().on_registers(\n",
+    "            x=(A[m - 1], A[-1]), y=(B[m - 1], B[-1]), ancilla=anc\n",
+    "        )\n",
+    "        yield op\n",
+    "        adjoint.append(op**-1)\n",
+    "\n",
+    "    # Build Tree\n",
+    "    yield from build_tree(A, B)\n",
+    "\n",
+    "    # Add the final SingleQubitCompare\n",
+    "    anc = [cirq.NamedQubit(f'anc_{i+anc_cnt}') for i in range(2)]\n",
+    "    anc_cnt += 2\n",
+    "    op = cirq_ft.algos.SingleQubitCompare()(A[-1], B[-1], *anc)\n",
+    "    yield op\n",
+    "\n",
+    "    # Update result qubits.\n",
+    "    yield cirq.CNOT(anc[0], less_than)\n",
+    "    yield cirq.CNOT(B[-1], z)\n",
+    "    yield cirq.CNOT(anc[1], greater_than)\n",
+    "\n",
+    "    # Undo changes made to data and ancilla qubits.\n",
+    "    yield op**-1\n",
+    "    yield from reversed(adjoint)\n",
+    "\n",
+    "\n",
+    "P = cirq.NamedQubit.range(3, prefix='p_')\n",
+    "Q = cirq.NamedQubit.range(3, prefix='q_')\n",
+    "quantum_compare = cirq.Circuit(quantum_quantum_comparator(P, Q, z, less_than, greater_than))\n",
+    "quantum_compare"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Now we check individual possiblities.\n",
+    "# Note it's a bit slow to simulate the 64 possibilies.\n",
```

Comment:
this is too slow for inclusion by default. These notebooks are run via the CI and in doc build jobs. Put it behind a comment or flag or something